mne_bids/write.py

```diff
@@ -963,20 +964,15 @@ def write_raw_bids(raw, bids_path, events_data=None,
                        "raw.times should not have changed since reading"
                        " in from the file. It may have been cropped.")
 
-    bids_path = bids_path.copy()
```

Comment:
I would still make a copy and return a modified copy because this will cause undesired behavior if you have a bids_path to start, call `write_raw_bids` without needing/wanting to use the output fpath, and then expect to use the original bids_path again.

Reply:
A lot of mne functions modify in place without an option... I agree that this may be an issue but I wrote is that way to be consistent e.g. https://mne.tools/stable/generated/mne.io.Raw.html#mne.io.Raw.set_annotations

Reply:
Also https://mne.tools/stable/generated/mne.io.Raw.html#mne.io.Raw.drop_channels, returns the modified instance but also acts in place as far as I remember. I always found that annoying because if you return the modified instance, to me you shouldn't modify in place as well.

My preference might be to return nothing an just modify the path instance and make that clear to the user that their path is updated and so to pass a copy if needed.

Reply:
I suppose @agramfort and @hoechenberger can chime in here?

If we want to modify in place, then a lot of tests will need to pass in the copied version instead (that was the difficulty I found, so I went w/ making a copy :p).

Reply:
@alexrockhill MNE function should not operate inplace. It's class method that do. Please avoid edge effects by modifying input parameters in functions if you can.

Reply:
Sounds good, thanks for the clarification and good call @adam2392